src/main/java/se/jsannemo/spooky/compiler/ir/ToIr.java

```diff
@@ -48,6 +49,21 @@ public static IrProgram generate(Program p) throws ValidationException {
         return ctx.program;
     }
 
+    private static void initDecl(List<VarDecl> globals, IrContext ctx) throws ValidationException {
+        // Create special __init__ function which will run before main.
+        IrFunction function = new IrFunction();
+        function.returnSignature = IrType.VOID;
+        ctx.program.functions.put("__init__", function);
+
+
+        // Allocate stack space and create init code for all globals.
+        ctx.function = function;
+        for (VarDecl global : globals) {
+          varDecl(global, ctx);
+        }
+        ctx.function = null;
```

Comment:
Add comment here that the fall-through of the function is OK since main starts right afterwards